src/clients/authorization/Scopes.tsx

```diff
@@ -0,0 +1,285 @@
+import React, { useState } from "react";
+import { Link, useHistory } from "react-router-dom";
+import { useTranslation } from "react-i18next";
+import {
+  Button,
+  DescriptionList,
+  DescriptionListDescription,
+  DescriptionListGroup,
+  DescriptionListTerm,
+  PageSection,
+  ToolbarItem,
+} from "@patternfly/react-core";
+import {
+  ExpandableRowContent,
+  TableComposable,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+} from "@patternfly/react-table";
+
+import type ResourceRepresentation from "@keycloak/keycloak-admin-client/lib/defs/resourceRepresentation";
+import type ScopeRepresentation from "@keycloak/keycloak-admin-client/lib/defs/scopeRepresentation";
+import type PolicyRepresentation from "@keycloak/keycloak-admin-client/lib/defs/policyRepresentation";
+
+import { KeycloakSpinner } from "../../components/keycloak-spinner/KeycloakSpinner";
+import { PaginatingTableToolbar } from "../../components/table-toolbar/PaginatingTableToolbar";
+import { useAdminClient, useFetch } from "../../context/auth/AdminClient";
+import { useRealm } from "../../context/realm-context/RealmContext";
+import { MoreLabel } from "./MoreLabel";
+import { toScopeDetails } from "../routes/Scope";
+import { toCreateScope } from "../routes/NewScope";
+import { ListEmptyState } from "../../components/list-empty-state/ListEmptyState";
+import useToggle from "../../utils/useToggle";
+import { DeleteScopeDialog } from "./DeleteScopeDialog";
+
+type ScopesProps = {
+  clientId: string;
+};
+
+export type ExpandableScopeRepresentation = ScopeRepresentation & {
+  resources: ResourceRepresentation[] | undefined;
+  permissions: PolicyRepresentation[] | undefined;
```

Comment:
If a field is optional you can use the following shorthand.
```suggestion
  permissions?: PolicyRepresentation[];
```